src/trajectory.jl

```diff
@@ -517,35 +442,48 @@ Base.:*(d1::Termination, d2::Termination) = Termination(d1.dynamic || d2.dynamic
 isterminated(d::Termination) = d.dynamic || d.numerical
 
 """
-    Termination(s::SliceTS, nt::NUTS, H0::F, H′::F) where {F<:AbstractFloat}
+    Termination(s::SliceTS, tc, H0::F, H′::F) where {F<:AbstractFloat}
 
 Check termination of a Hamiltonian trajectory.
 """
-function Termination(s::SliceTS, nt::NUTS, H0::F, H′::F) where {F<:AbstractFloat}
-    return Termination(false, !(s.ℓu < nt.Δ_max + -H′))
+function Termination(s::SliceTS, tc, H0::F, H′::F) where {F<:AbstractFloat}
+    return Termination(false, !(s.ℓu < tc.Δ_max + -H′))
 end
 
 """
-    Termination(s::MultinomialTS, nt::NUTS, H0::F, H′::F) where {F<:AbstractFloat}
+    Termination(s::MultinomialTS, tc, H0::F, H′::F) where {F<:AbstractFloat}
 
 Check termination of a Hamiltonian trajectory.
 """
-function Termination(s::MultinomialTS, nt::NUTS, H0::F, H′::F) where {F<:AbstractFloat}
-    return Termination(false, !(-H0 < nt.Δ_max + -H′))
+function Termination(s::MultinomialTS, tc, H0::F, H′::F) where {F<:AbstractFloat}
+    return Termination(false, !(-H0 < tc.Δ_max + -H′))
 end
 
 """
 A full binary tree trajectory with only necessary leaves and information stored.
 """
-struct BinaryTree{C<:AbstractTerminationCriterion}
+struct BinaryTree{R<:Union{Nothing, AbstractVecOrMat}}
     zleft   # left most leaf node
     zright  # right most leaf node
-    c::C    # termination criterion
+    rho::R  # termination statistic
     sum_α   # MH stats, i.e. sum of MH accept prob for all leapfrog steps
     nα      # total # of leap frog steps, i.e. phase points in a trajectory
     ΔH_max  # energy in tree with largest absolute different from initial energy
 end
 
+"Initialize termination statistic"
+rho_init(::ClassicNoUTurn, z) = nothing
```

Comment:
Consider introducing a `TurnStatistic` here. Then `rho_init` can be replaced with a `TurnStatistic` constructor overloaded for various turn termination types, `rho_merge` can be replaced with `combine(::TurnStatistic, ::TurnStatistic)`.